CRM/Report/Form.php

```diff
@@ -2637,23 +2643,22 @@ public function buildQuery($applyLimit = TRUE) {
    * Build group by clause.
    */
   public function groupBy() {
-    $groupBys = array();
     if (!empty($this->_params['group_bys']) &&
       is_array($this->_params['group_bys'])
     ) {
       foreach ($this->_columns as $tableName => $table) {
         if (array_key_exists('group_bys', $table)) {
           foreach ($table['group_bys'] as $fieldName => $field) {
             if (!empty($this->_params['group_bys'][$fieldName])) {
-              $groupBys[] = $field['dbAlias'];
+              $this->_groupByArray[] = $field['dbAlias'];
```

Comment:
@totten @eileenmcnaughton i have cherry-picked this commit into my other PR and made some associated changes. Tim i think it would be better viewed in this PR https://github.com/civicrm/civicrm-core/issues/10765 so you can see how it would get applied to fix the php7.1 errors